app/services/versioned_files_service/migrate_action.rb

```diff
@@ -0,0 +1,58 @@
+class VersionedFilesService
+  # Migrate an object from unversioned to versioned Stacks layout.
+  class MigrateAction
+    def initialize(service:, version_metadata:)
+      @service = service
+      @version_metadata = version_metadata
+    end
+
+    def call
+      # For each shelved file in the cocina object, make sure there is a content file.
+      check_content_files!
+      # For each shelved file, create a hardlink named by md5
+      link_content_files
+      # Write the cocina to cocina path for the version and create a new head cocina symlink.
+      write_cocina(version: '1', cocina: cocina_hash, head_version: true)
+      # Write the public xml to public xml path for the version and create a new head public xml symlink .
+      write_public_xml(version: '1', public_xml:, head_version: true)
+      # Update the version manifest.
+      version_manifest.update_version(version: '1', version_metadata:, head_version: true)
+    end
+
+    private
+
+    attr_reader :version_metadata
+
+    delegate :purl_object_path, :content_path, :content_path_for, :stacks_object_path,
+             :write_cocina, :write_public_xml, :version_manifest, to: :@service
+
+    def check_content_files!
+      shelve_file_map.each_key do |filename|
+        raise Error, "Content file for #{filename} not found" unless stacks_content_path_for(filename:).exist?
+      end
+    end
+
+    def link_content_files
+      FileUtils.mkdir_p(content_path)
+      shelve_file_map.each do |filename, md5|
+        LinkSupport.link(stacks_content_path_for(filename:), content_path_for(md5:))
+      end
+    end
+
+    def shelve_file_map
+      @shelve_file_map ||= Cocina.new(hash: cocina_hash).shelve_file_map
```

Comment:
Can this be a method in the `Cocina` class (e.g. `Cocina.for_version(version:)`)

Reply:
Well, took me trying to implement to realize this, but: This is loading the cocina from the PURL file system. There is no version involved and I think this is the only place where we would want to do this.